src/frontend/site.js

```diff
@@ -328,16 +341,65 @@ function postBio(req, res) {
         .then((_) => {
             res.redirect(extendURL(`/user/${getJwtUser(req.cookies)}`));
         }).catch(error => {
-            res.status(statusCodeForError(error)).render('error.njk', handleError(error));
-        });
+        res.status(statusCodeForError(error)).render('error.njk', handleError(error));
+    });
 }
 
-/**
+function postLike(req, res) {
+    const postId = req.params.postId;
+    fetchUsingDeploymentBase(req, () => req.LIKE_SERVICE_API.get(`/like-service/like-count/` + postId)
+    ).then((response => {
+        let likeData = response.data;
+        if (likeData.userLiked) {
+            console.log(postId)
+            fetchUsingDeploymentBase(req, () => req.LIKE_SERVICE_API.post(`/like-service/like-delete`, {postId: postId})).then(response => {
+                res.redirect(extendURL('/post/' + postId))
+            });
+        } else {
+            fetchUsingDeploymentBase(req, () => req.LIKE_SERVICE_API.post(`/like-service/like-post`, {postId: postId}
+            )).then((response => {
+                res.redirect(extendURL('/post/' + postId))
+            }))
+        }
+    }))
+}
+
+
+async function getLikeCount(req, postId) {
```

Comment:
[DISCUSS] I think you could remove the outer promise:
```js
async function getLikeCount(req, postId) {
    let response = await fetchUsingDeploymentBase(req, () => req.LIKE_SERVICE_API.get(`/like-service/like-count/` + postId))
    return response.data
}
```

Reply:
Outer promise removed